Step3. [자동차경주] 2차 리뷰요청 드립니다.

AyoungLim1103 opened the pull request:
Step3. [자동차경주] 리뷰요청 드립니다.

Commit:
feat(racingCar) : car ready and start race event

added new event to racingCar :
- ready car
- start race car

Commit:
feat(racingCar) : car ready and start race event

added new event to racingCar :
- ready car
- start race car

Commit:
feat(racingCar) : car ready and start race event

added new event to racingCar :
- ready car
- start race car

Commit:
feat(racingCar) : car ready and start race event

added new event to racingCar :
- ready car
- start race car

gw-study commented on src/main/java/racingcar/Main.java:
```diff
@@ -0,0 +1,13 @@
+package racingcar;
+
+public class Main {
```
![image](https://user-images.githubusercontent.com/84048349/121998654-26619180-cde7-11eb-9ca1-e60b4d8165c5.png)
실행결과가 기대한 값이 아닌거 같습니다ㅠ

gw-study commented on src/main/java/racingcar/InputView.java:
```diff
@@ -0,0 +1,26 @@
+    private int carCount;
+    private int tryCount;
+
+    public void getEventInfo(){
```
`inputCarRacingEvent` 정도로 메소드명을 수정해 보는게 어떨까요?
통상적으로 `getXXX` 처럼 게터는 어떤 값을 리턴하기를 기대합니다.
그리고 XXXInfo 라는 이름은 Info라는 이름이 너무 광범위해서 클래스 또는 메서드명에
적합하지 않을 수 있어요😱
가능하면 메서드 이름을 보고 어떤 행동을 할지 쉽게 유추가 가능하면 좋을거 같아요.
(한국사람이라서 제일 힘들어 하는것 중에 하나이긴 합니다😁)

gw-study commented on src/main/java/racingcar/Racing.java:
```diff
@@ -0,0 +1,39 @@
+        }
+    }
+
+    public void startRace(int tryCount){
```
출력을 하는 행위는 UI패키지인 ResultView에 위임하는게 결합도를 느슨하게 할 수 있을거 같아요.
도메인 클래스인 `Racing`에서는 View 패키지의 클래스를 참조하지 않도록
변경해 볼 수 있을까요?

gw-study commented on src/main/java/racingcar/Racing.java:
```diff
@@ -0,0 +1,39 @@
+        }
+    }
+
+    public void moveCars(){
```
위 코멘트 내용과 동일한 의견입니다

gw-study commented on src/main/java/racingcar/Forward.java:
```diff
@@ -0,0 +1,13 @@
+public class Forward {
+    private Random random = new Random();
+
+    static final Integer TEN = 10;
```
접근지정자는 필수로 기입 부탁드립니다🙏
그리고 Integer로 선언한 이유가 있을까요?🤔
int로도 충분해 보이는데 다른 의도가 있으실까요?

gw-study commented on src/main/java/racingcar/ResultView.java:
```diff
@@ -0,0 +1,13 @@
+        System.out.println("실행결과");
+    }
+
+    private static final char TRACKSHAPE = '-';
```
`final`사용 잘 하셨습니다👍

gw-study reviewed (changes requested):
코드가 깔끔하고 간결하게 작성 잘 해주셨네요👍👍
커밋 메세지 작성도 아주 잘 해주셨어요💯

자동차 경주 프로그램을 실행했을 때 기대한 결과가
나타나지 않는 부분과 도메인과 출력 로직 분리를 위한 코멘트를 몇개 남겨놓았어요😅

확인해보시고 다시한번 리뷰 요청 부탁드려요~🙇‍♂️
궁금한 내용은 언제든지 DM보내주세요😀

아!그리고 `README.md`파일에 기능목록 정리하는 요구사항이 빠져있는거 같아요.
같이 확인 부탁드리겠습니다.🙇‍♂️🙇‍♂️

Commit:
fix(racingCar) : fix Racing Object and ResultView

- fix moveCars function (remove calling resultView acton)
- modify showRacingCarResult function in ResultView
- some of variables..
- the name of function in InputView.

AyoungLim1103 committed:
Update README.md

AyoungLim1103 commented:
Step3 [자동차경주] 2차 리뷰 요청드립니다. 감사합니다 :)

gw-study commented on README.md:
```diff
@@ -6,3 +6,9 @@
 * 코드 리뷰 피드백에 대한 개선 작업을 하고 다시 PUSH한다.
 * 모든 피드백을 완료하면 다음 단계를 도전하고 앞의 과정을 반복한다.
+
+## 기능 구현
+1. 자동차경주를 시작할 자동차 대수를 input 한다.(Scanner 사용)
+2. 자동차가 전진할 횟수를 input 한다 (Scanner 이용)
+3. 전진 여부는 random 함수를 이용하여 random value 가 4이상 9이하일 경우만 전진하도록 설정한다.
+4. 결과는 '-' (대시) 를 통해 보여준다.
```
기능목록 정리👍

gw-study commented on src/main/java/racingcar/Car.java:
```diff
@@ -0,0 +1,22 @@
+        return currentPosition;
+    }
+
+    public String showTrack(char trackShape) {
+        StringBuffer track = new StringBuffer();
```
`showTrack`도 출력을 위한 로직 같아 보입니다😅
View로 이동 부탁드려요~

gw-study commented on src/main/java/racingcar/Main.java:
```diff
@@ -0,0 +1,17 @@
+package racingcar;
+
+public class Main {
+    public static void main(String[] args) {
+
```
`main`함수가 컨트롤러 역할을 잘 수행해주고 있네요👏👏👏

gw-study commented on src/main/java/racingcar/Racing.java:
```diff
@@ -0,0 +1,27 @@
+        return this.cars;
+    }
+
+    public void readyCar(int CarCount){
+        for(int i = 0; i < CarCount; i++){
```
앗! 파라메터 변수는 소문자로 시작해야합니다!!😅

gw-study commented on src/main/java/racingcar/ResultView.java:
```diff
@@ -0,0 +1,18 @@
+
+    public static void showRacingCarResult(List<Car> cars) {
+        System.out.println("");
+        for(Car car : cars) {
+            System.out.println(car.showTrack(TRACKSHAPE));
```
View에서 도메인(Model) 참조는 지양해 주세요.
앞에서 코멘트 드린것처럼 `showTrack`의 역할은 View에서 수행해야 할 거 같네요😁

아!코멘트 수정할께요
View에서 도메인(Model) 참조는 해주셔도 됩니다. 반대가 불가구요ㅎ
 `showTrack`의 역할만 View쪽으로 이동 부탁드릴께요🙏

gw-study commented on src/test/java/racingcar/RacingCarTest.java:
```diff
@@ -0,0 +1,39 @@
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class RacingCarTest {
+    Racing racing;
```
'1대의 자동차를 3번 반복하면 3칸을 전진한다.' 라는 테스트도 수행해 볼 수 있을까요?

랜덤 함수를 사용하기 때문에 이 테스트가 쉽지 않을 수 있어요.
그래도 어떻게 하면 가능할지에 대해 고민해 보시면 많은 걸 얻어가실 수 있을거예요😃
 * 힌트 : 자바의 Interface 를 생각해보세요

gw-study reviewed (approved):
피드백 적용 잘해주셨어요👍👍👍
저번에도 말씀 드렸는데 코드가 간결해서 읽기가 참 수월해요 :)

몇가지 코멘트를 드린게 있는데요,
다음 단계에서 확인 부탁드리겠습니다.

이번 단계는 여기서 머지 하도록 할께요
다음 단계도 퐈이팅~🔥🔥🔥
